### PR TITLE
Add user login state original traits and indicator that original values set.

### PR DESCRIPTION
**Pull request by mdwn**
Original traits has been added to the user login state for use by a watcher in the enterprise repo that will react to user access changes. This field will be used to rebuild the last known good state when user access changes.

Additionally, an internal annotation that indicates the original role/trait fields have been set in the user login state has been introduced so that the watcher knows whether the user login state can be used by this watcher.

### CODE REVIEW

**Review by fspmarshall — APPROVED** [review]
Code looks reasonable.  This does make me wonder though... what do you think about just embedding the actual user spec inside the login state instead and just doing something like GetOriginalSpec() types.User or w/e?  Then when user resource gets updated the login state inherits the new state for free.  Might save future pain.  Just a thought, feel free to take it or leave it.

**Comment by mdwn** [comment]
> Code looks reasonable. This does make me wonder though... what do you think about just embedding the actual user spec inside the login state instead and just doing something like GetOriginalSpec() types.User or w/e? Then when user resource gets updated the login state inherits the new state for free. Might save future pain. Just a thought, feel free to take it or leave it.

I'm going to punt on this for now. I think it's a good idea, but it sort of introduces some uncomfortable questions w/r/t gogo vs. protobuf interactions.

File: api/types/userloginstate/user_login_state.go [hunk 1]
```diff
@@ -25,6 +25,14 @@ import (
 	"github.com/gravitational/teleport/api/types/trait"
 )
 
+const (
+	// OriginalRolesAndTraitsSet is an annotation that will indicate that the original roles and traits have been
+	// set for this objects. For existing user login states, this will not be set, indicating that we can't
+	// use these fields reliably.
+	// TODO(mdwn): Remove this in v17.
```
Comment (AntonAM) [inline api/types/userloginstate/user_login_state.go:32]: We usually use a bit different form for this: `// DELETE IN 17 (mdwn)` , it's easier to find them all just by search if they are standardized.

**Review by AntonAM — APPROVED** [review]

File: api/types/userloginstate/user_login_state.go (hunk 1 above)
Comment (mdwn) [reply api/types/userloginstate/user_login_state.go:32]: Will do!

**Comment by strideynet** [comment]
Likely introduced https://github.com/gravitational/teleport/issues/35443

**Pull request merged**